src/Observers/RevisionableObserver.php

```diff
@@ -100,6 +98,12 @@ public function deleting($model)
     public function deleted($model)
     {
         if (!method_exists($model, 'bootSoftDeletes') || $model->isForceDeleting()) {
+            $revision = $model->revision;
+            // if newer revision exists, point its previous_revision_id to the previous revision of this item
+            if ($revision !== null && $revision->next()->exists()) {
+                $revision->next->previous_revision_id = $revision->previous_revision_id;
+                $revision->next->save();
+            }
```

Comment:
Just noting that this might be superseded by the changes coming to fix #2, more as a note to myself